nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/Instrumenter.java

```diff
@@ -87,17 +139,57 @@ private static void instrumentClass(String className,
         outZip.closeEntry();
     }
 
-    public static final void instrument(String rtJar, String outJar) throws IOException {
+    private static void filterCachedFrom(HashSet<String> toShuffle, HashSet<String> toCopy,
+                                         ZipFile rt, ZipFile outJarZipFile)
+            throws IOException, NoSuchAlgorithmException {
+        Iterator<String> it = toShuffle.iterator();
+        while (it.hasNext()) {
+            String cl = it.next();
+            InputStream clInputStream = rt.getInputStream(rt.getEntry(cl));
+            InputStream md5InputStream = outJarZipFile.getInputStream(outJarZipFile.getEntry(cl + ".md5"));
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The String literal ".md5" appears 5 times in this file; the first occurrence is on line 149](https://www.codacy.com/app/gyori/NonDex/file/2716321334/issues/source?bid=3322087&fileBranchId=3343849#l149)